app/models/declaration/submissions/SubmissionStatus.scala

```diff
@@ -85,56 +85,77 @@ object Status {
       case "18"                                                                  => AwaitingExitResults
       case _                                                                     => UnknownStatus
     }
+
+  def retrieve(functionCode: String, nameCode: Option[String]): SubmissionStatus =
```

Comment:
If SubmissionStatus had a value method (or something) that had to be overridden we could set the “value” to the below

Reply:
I agree and I tried it in the morning. Complication appears around 2 statuses which require nameCode field. This would require additional thought. And if the solution is found, the backend should be updated as well.

Reply:
Found working solution for this now. The commit follows.